crates/spirv-std/src/arch.rs

```diff
@@ -188,6 +188,7 @@ pub unsafe fn read_clock_uvec2_khr<V: Vector<u32, 2>, const SCOPE: u32>() -> V {
 }
 
 #[spirv_std_macros::gpu_only]
+#[allow(dead_code)]
```

Comment:
```suggestion
#[cfg(target_arch = "spirv")]
```

(more details in the other comment, but this might just work, and it's a clean local change)